src/main/java/com/telefonica/iot/cygnus/sinks/OrionHDFSSink.java

```diff
@@ -218,65 +218,29 @@ protected void setPersistenceBackend(HDFSBackendImplREST persistenceBackend) {
 
     @Override
     public void configure(Context context) {
-        String cosmosHost = context.getString("cosmos_host");
-        String hdfsHost = context.getString("hdfs_host");
-
-        if (hdfsHost != null && hdfsHost.length() > 0) {
-            host = hdfsHost.split(",");
-            LOGGER.debug("[" + this.getName() + "] Reading configuration (hdfs_host=" + Arrays.toString(host) + ")");
-        } else if (cosmosHost != null && cosmosHost.length() > 0) {
-            host = cosmosHost.split(",");
-            LOGGER.debug("[" + this.getName() + "] Reading configuration (cosmos_host=" + Arrays.toString(host) + ")"
-                    + " -- DEPRECATED, use hdfs_host instead");
-        } else {
-            host = new String[]{"localhost"};
-            LOGGER.debug("[" + this.getName() + "] Defaulting to hdfs_host=localhost");
-        } // if else
-
-        String cosmosPort = context.getString("cosmos_port");
-        String hdfsPort = context.getString("hdfs_port");
-        int intPort;
-
-        if (hdfsPort != null && hdfsPort.length() > 0) {
-            port = hdfsPort;
-            intPort = Integer.parseInt(port);
+        String hdfsHost = context.getString("hdfs_host", "localhost");
+        host = hdfsHost.split(",");
+        LOGGER.debug("[" + this.getName() + "] Reading configuration (hdfs_host=" + Arrays.toString(host) + ")");
+        
+        port = context.getString("hdfs_port", "14000");
+        if (!(port.equals(""))) {
```

Comment:
I think we can make the checks for the ports stronger. First, by checking if it is a number:

```
port = context.getString("hdfs_port", "14000");

// check if the configured port is a number
try {
   int intPort = Integer.parseInt(port);
} catch (Exception e) {
   invalidConfiguration = true;
   LOGGER.debug("Invalid configuration ... ");
} // try catch

// check if the number is a valid port number
if (!invalidConfiguration && intPort > 65535) {
   invalidconfiguration = true;
   LOGGER.debug("Invalid configuration ... ");
} else {
   LOGGER.debug("Reading configuration ... ");
} // if else
```